File: leetcode.js/两数之和.js
var twoSum = function(nums, target) {
    const hashMap = {};
    for(i = 0; i < nums.length; i++){
        const com = target - nums[i];
        if(com in hashMap){
            return [hashMap[com], i];
        }
        hashMap[nums[i]] = i
    }

};
const nums = [2,2,2,15];
const target = 17;
console.log(twoSum(nums, target));

File: leetcode.py/两数之和.py
class Solution(object):
    def twoSum(self, nums, target):
        seen = {}
        for i,val in enumerate(nums):
            need = target - val
            if need in seen:
                return [seen[need], i]
            seen[val] = i
        return []

            
nums = [2,7,11,15]
target = 9
s = Solution()
print(s.twoSum(nums, target))
